### DISCUSSION TITLE
Using Repeater according to the tutorial, I don't know why an error was reported in Array to string conversion. Although I didn't use the "I offer" and "socialbutton" methods in SocialOffer, I thought they could establish a many to many relationship, but an error was reported. TextInput:: make ('type ') cannot set the specified document and it seems that it hasn't been written

### DISCUSSION THREAD

**Discussion by YANGJIAXUE2022510** [filamentphp/filament#8429] [General]
Using Repeater according to the tutorial, I don't know why an error was reported in Array to string conversion. Although I didn't use the "I offer" and "socialbutton" methods in SocialOffer, I thought they could establish a many to many relationship, but an error was reported. TextInput:: make ('type ') cannot set the specified document and it seems that it hasn't been written
![8545ab10c095f072f8f4a96b29f654a](https://github.com/filamentphp/filament/assets/105269516/ec3c4f74-0e43-4aa8-ac00-5a1001e6953e)
![c641ec54ce2dfabc20680edf759772f](https://github.com/filamentphp/filament/assets/105269516/adeeb847-c8a6-4d51-a74e-b13409e356d8)
![8bd77ca3b33ca6d381301343b4a05b1](https://github.com/filamentphp/filament/assets/105269516/5a514043-87a8-46b5-a997-eaebfa2d971d)
![3d7e12ac1f48c0d06d952a12148ba8d](https://github.com/filamentphp/filament/assets/105269516/1b4f39be-08f0-4441-9601-003f8c1b95df)
![4e108970b9097979fa3fd54ec5dcec4](https://github.com/filamentphp/filament/assets/105269516/c1121506-5349-4517-9757-8fef38e58de1)
![4f1cecde1fbff6765f5d9cc6a3891ff](https://github.com/filamentphp/filament/assets/105269516/d55a6ac9-5470-48da-96ef-656343e9fafd)

**Comment by leandrocfe** [comment]
I think your repeater should be
```php
Repeater::make('socialoffer')
    ->relationship()
    ->schema([
        Select::make('social_button_id')
            ->relationship('socialbutton', 'buttontext')
            ->required()
    ]),
```
If you want to add the social button, you can use [creating-a-new-option-in-a-modal](https://filamentphp.com/docs/3.x/forms/fields/select#creating-a-new-option-in-a-modal)

And the name should be `offer_social`
[many-to-many-table-structure](https://laravel.com/docs/10.x/eloquent-relationships#many-to-many-table-structure)

**Reply by YANGJIAXUE2022510** [reply]
But if I write this way, I will also report an error in Array to string conversion



How to make money by embedding TextInput into the Repeater I want to use

Is my writing style correct?
```
Repeater::make('socialoffer')
    ->relationship()
    ->schema([
        TextInput::make('type')->label('类型'),
        Select::make('social_button_id')
            ->relationship('socialbutton', 'buttontext')
            ->required()
    ]),
```

**Reply by leandrocfe** [reply]
Does it work if you remove TextInput?

**Reply by YANGJIAXUE2022510** [reply]
"Yes, I still get the 'Array to string conversion' error.

Repeater indicates that I'm trying to store this array as a string.
```Repeater::make('socialoffer')
     ->schema([
     // TextInput::make('type')->label('Type'),
     Select::make('social_button_id')
     ->relationship('socialbutton', 'buttontext')
     ->required()
    ])
```
Even though I'm passing an array to the Repeater.

However, in my Offer.php file, I have written:
```    
public function socialbutton(): HasMany
    {
        return $this->hasMany(SocialButton::class);
    }
```
The error might be in the 'Repeater::make('socialoffer')' part, specifically in the 'socialoffer' method.

I've defined the method like this:
```  
 public function socialoffer(): HasMany
    {
        return $this->hasMany(SocialOffer::class);
    }
```
Since it's necessary to have the 'socialoffer' method, I suspect there might be an issue with how I've defined it."

**Reply by leandrocfe** [reply]
Can you share the repo on github?
